schemas/definitions.json

```diff
@@ -0,0 +1,78 @@
+{
+  "$schema": "http://json-schema.org/draft-04/schema#",
+  "id": "definitions.json#",
+  "title": "Schema for common fields in data files",
+  "definitions": {
+    "faction": {
+      "type": "string",
+      "description": "A faction in the game.",
+      "enum": [
+        "First Order",
+        "Galactic Empire",
+        "Rebel Alliance",
+        "Resistance",
+        "Scum and Villainy"
+      ]
+    },
+    "size": {
+      "type": "string",
+      "description": "A ship size in the game.",
+      "enum": [
+        "huge",
+        "large",
+        "small"
+      ]
+    },
+    "slot": {
+      "type": "string",
+      "description": "A slot type in the game.",
+      "enum": [
+        "Astromech",
+        "Bomb",
+        "Cannon",
+        "Cargo",
+        "Crew",
+        "Elite",
+        "Hardpoint",
+        "Illicit",
+        "Missile",
+        "Modification",
+        "Salvaged Astromech",
+        "System",
+        "Team",
+        "Tech",
+        "Title",
+        "Torpedo",
+        "Turret"
+      ]
+    },
+    "action": {
+      "type": "string",
+      "description": "A action type in the game.",
+      "enum": [
+        "Barrel Roll",
+        "Boost",
+        "Cloak",
+        "Coordinate",
+        "Evade",
+        "Focus",
+        "Jam",
+        "Recover",
+        "Reinforce",
+        "Rotate Arc",
+        "SLAM",
+        "Target Lock"
+      ]
+    },
+    "file_path": {
+      "type": "string",
+      "description": "A file path",
+      "pattern": "^[a-zA-Z\\(\\)\\.0-9_-]([a-zA-Z\\(\\)\\.0-9\\ _-]*[a-zA-Z\\(\\)\\.0-9\\ _-])?(\\/[a-zA-Z\\(\\)\\.0-9_-]([a-zA-Z\\(\\)\\.0-9\\ _-]*[a-zA-Z\\(\\)\\.0-9\\ _-])?)*.[a-z0-9]{3}$"
```

Comment:
As we're only using this for images, I'd rather see a regex that validates proper image paths instead of _any_ file path.